src/pykeen/nn/message_passing.py

```diff
@@ -364,9 +365,10 @@ def forward_vertically_stacked(self, x: torch.Tensor, adj: torch.Tensor) -> torc
         # (R * n, di) -> (R, n, nb, bsi)
         x = x.view(self.num_relations, -1, self.num_blocks, self.input_block_size)
         # (R, nb, bsi, bso), (R, n, nb, bsi) -> (n, nb, bso)
-        x = torch.einsum("rbio, rnbi -> nbo", self.blocks, x)
+        x = einsum("rbio, rnbi -> nbo", self.blocks, x)
         # (n, nb, bso) -> (n, do)
-        x = x.view(x.shape[0], self.num_blocks * self.output_block_size)
+        # note: depending on the contracting order, the output may supporting viewing, or not
```

Comment:
does that mean that given some inputs this will fail?

Reply:
No. `reshape` always works, and will also use a `view` whenever possible.

In cases where `view` does not work (but `reshape` does), the operation will be more expensive, since the operation has to restore contiguity (by potentially re-copying data).

Reply:
[here](https://pytorch.org/docs/stable/generated/torch.Tensor.view.html) is a longer description thereof.

Reply:
okay, up to you if you want to update that docstring or not, then :shipit: